drift_dev/lib/src/lints/unawaited_futures_in_transaction.dart

```diff
@@ -0,0 +1,156 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/error/error.dart' hide LintCode;
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+
+final tableChecker = TypeChecker.fromName('Table', packageName: 'drift');
+final databaseConnectionUserChecker =
+    TypeChecker.fromName('DatabaseConnectionUser', packageName: 'drift');
+final columnBuilderChecker =
+    TypeChecker.fromName('ColumnBuilder', packageName: 'drift');
+
+class UnawaitedFuturesInTransaction extends DartLintRule {
+  UnawaitedFuturesInTransaction() : super(code: _code);
+
+  static const _code = LintCode(
+    name: 'unawaited_futures_in_transaction',
+    problemMessage:
+        'All futures in a transaction should be awaited to ensure that all operations are completed before the transaction is closed.',
+    errorSeverity: ErrorSeverity.ERROR,
+  );
+  @override
+  void run(CustomLintResolver resolver, ErrorReporter reporter,
+      CustomLintContext context) {
+    context.registry.addExpressionStatement((node) {
+      node.accept(_Visitor(this, reporter, _code));
+    });
+    context.registry.addCascadeExpression((node) {
+      node.accept(_Visitor(this, reporter, _code));
+    });
+    context.registry.addInterpolationExpression((node) {
+      node.accept(_Visitor(this, reporter, _code));
+    });
+  }
+}
+
+// Copyright (c) 2016, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
```

Comment:
Let's also link to the unawaited futures link source where I guess this is coming from

Reply:
Done